CSharp/DevVmPowershell/Helpers/Implementations/InstanceSettingsHelper.cs

```diff
@@ -125,5 +129,37 @@ public string GetInstanceSettingValue(string name, string section)
 				throw new Exception("Failed to Get Instance Setting Value", ex);
 			}
 		}
+
+		public int GetInstanceSettingArtifactIdByName(string name, string section)
+		{
+			try
+			{
+				int artifactId = 0;
+				using (IInstanceSettingManager instanceSettingManager = ServiceFactory.CreateProxy<IInstanceSettingManager>())
+				{
+					Query query = new Query();
+					query.Condition = $"'Section' == '{section}' AND 'Name' == '{name}'";
```

Comment:
It might not be necessary but I know you can use TextConditions to build a query string, but you'll have to do something like .ToQueryString() at the end.  Might be overkill for this but just a thought